test/Generator/MethodGeneratorTest.php

```diff
@@ -307,6 +307,37 @@ public function testCreateFromArray()
         self::assertSame('\\SampleType', $methodGenerator->getReturnType()->generate());
     }
 
+    /**
+     * @dataProvider returnReturnsReferenceValues
+     * @param bool|string|int $value
+     * @param bool $expected
+     */
+    public function testCreateFromArrayWithReturnsReference($value, $expected): void
+    {
+        $methodGenerator = MethodGenerator::fromArray([
+            'name'             => 'SampleMethod',
+            'returnsreference' => $value,
+        ]);
+
+        self::assertSame($expected, $methodGenerator->isReturnsReference());
+    }
+
+    /**
+     * @return string[][]
```

Comment:
Just realized this type is wrong as well (copied from somewhere else and then forgotten about), though not sure about conventions in the project as to what I should change it to.

Reply:
I'll adjust this on merge :+1: